### ISSUE TITLE
Events of the same type that end simultaneously 

### ISSUE THREAD

**Issue by rhodricusack** [nilearn/nilearn#2674]
nilearn 0.7.0

Hi-
I've accidentally discovered a new way to break make_first_level_design_matrix, with overlapping events of the same trial_type but different durations. If two events like this end simultaneously, only one of them gets ended, leading to an elevated regressor for the rest of the frames.

Below is some example code, and the figure it produces:
![break_make_first_level_design_matrix](https://user-images.githubusercontent.com/1569737/106321615-6f67dd80-626c-11eb-95bb-5aa1c8ae943f.jpg)

It took a while to find this one! I think the reason I'm coming across these is that I'm modelling naturalistic stimuli using ratings, and using boxcars rather than creating regressors.
Very best
Rhodri

```python
import pickle
import pandas as pd
from nilearn.glm.first_level import make_first_level_design_matrix
import numpy as np
from matplotlib import pyplot as plt

frame_times=np.arange(20)

fig, ax= plt.subplots(nrows=2)

# All good
events=pd.DataFrame({'onset': [2, 4, 4.5], 'duration': [1.0, 1.0, 1.0], 'trial_type': ['vanilla']*3})
X = make_first_level_design_matrix(frame_times, events= events, hrf_model=None)
ax[0].plot(X['vanilla'])

# Second and third events end at same time, so second event never ends!
events=pd.DataFrame({'onset': [2, 4, 4.5], 'duration': [1.0, 1.5, 1.0], 'trial_type': ['vanilla']*3})
X = make_first_level_design_matrix(frame_times, events= events, hrf_model=None)
ax[1].plot(X['vanilla'])

plt.savefig('break_make_first_level_design_matrix.jpg')
```

**Comment by tsalo** [comment]
I just ran your code locally using the current master branch and it looks different to me:

![image](https://user-images.githubusercontent.com/8228902/106323743-e76edd00-6245-11eb-8d44-4dd8d9570dde.png)

Are you able to reproduce the bug on the current master branch?

**Comment by rhodricusack** [comment]
Yes, installing the current master branch fixes the problem for me too.
Thank you!

________________________________

**Comment by tsalo** [comment]
Since this will be fixed in the 0.7.1 release, I'm going to close it. For posterity- any users who come across this issue should update `nilearn` to >= 0.7.1 (or the `master` branch if 0.7.1 hasn't been released yet).

I'll also open a PR to update [`whats_new.rst`](https://github.com/nilearn/nilearn/blob/master/doc/whats_new.rst) with information about the bug. Thanks @rhodricusack for reporting the bug- otherwise, I'm not sure it would have been documented.
